File: storage/storage-mysql/src/main/java/io/say/better/storage/mysql/dao/repository/LearnerReadRepository.java
package io.say.better.storage.mysql.dao.repository;

import io.say.better.core.common.Provider;
import io.say.better.storage.mysql.domain.entity.Learner;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface LearnerReadRepository extends JpaRepository<Learner, Long>, MemberReadRepository<Learner> {

	Optional<Learner> findByProviderAndLoginId(Provider provider, String loginId);

	Optional<Learner> findByEmail(String email);
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/dao/repository/LearnerReadRepository.kt
package io.say.better.storage.mysql.dao.repository

import io.say.better.storage.mysql.domain.entity.Learner
import io.say.better.storage.mysql.domain.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface LearnerReadRepository : JpaRepository<Learner, Long> {
    fun findByLearnerId(learnerId: Long): Optional<Learner>
    fun findByMemberId(learnerMember: Member): Optional<Learner>
}
